### ISSUE TITLE
[LOW] Distance: Handle errors with geocoding for offline distance requests

### ISSUE THREAD

**Issue by neil-marcellini** [Expensify/App#22719]
Check out the original plan in the design doc: [Handling errors](https://docs.google.com/document/d/16Q9sE-991CZ4wUcalOs_U-qyEEza_C9tZwj6G5lHUp0/edit#heading=h.qdqtogd0n2t4) with geocoding.

Since then, we've updated the desired UX as outline in [this Slack conversation](https://expensify.slack.com/archives/C05DWUDHVK7/p1699392640830819). Here's the current desired UX.

Here is how the flow currently works
1. Go offline
2. Request money > Distance
3. Enter invalid start and finish addresses
4. Click next
5. Choose a workspace
6. Click request
7. Go online
8. Lots of issues

Here's how it should work after going online

- The problematic request appears in the LHN as unread, with a red dot indicator. This matches the [wave9 violations UX](https://docs.google.com/document/d/1iGvpCoQ6Wt-i2wcpvaMKsWZRNs-oAsWoaS_nLue7_E0/edit#heading=h.k7b6tjysjfjf).
- There is a red dot on the distance row on the transaction report page
- Going back up the hierarchy, red dot also appears on the request preview in the expense report, and on the expense report preview in the workspace chat
- We must be able to open all reports that potentially failed to create. Workspace chat, expense report, distance transaction report
- After clicking on the distance row you are navigated to the distance request page and error messages are displayed under each invalid waypoint, instructing the user to edit them. When a waypoint is clicked the user can update the address and must use the suggestions which are now shown since they are online
- When a new address is saved we clear any errors on the waypoint
- When the save button is clicked we first make sure that all waypoints errors have been cleared, then call CreateDistanceRequest to try creating the request again instead of editing an existing request. We can identify that the request hasn’t been created yet because it doesn’t have a receipt.
❗ Note, this feels a bit sketchy. Is this going to work? Are we ok with the data only existing optimistically? Can we open all the chats without the request existing?
- When the request is successfully created, clear all errors and make sure all optimistic data is updated or replaced. 
- Prior to retrying creation, the error messages can not be dismissed. If the user wants to give up they can delete the optimistic request with the normal delete flow and we must clean up all optimistic data.

**Comment by neil-marcellini** [comment]
On hold for [Fix error handling when creating a request fails](https://github.com/Expensify/App/issues/22126#top) because we'll need to use a lot of that functionality.

**Comment by neil-marcellini** [comment]
Still holding

**Comment by neil-marcellini** [comment]
Still holding, making some progress on that issue.

**Comment by neil-marcellini** [comment]
Still on hold

**Comment by neil-marcellini** [comment]
ditto

**Comment by neil-marcellini** [comment]
I'm taking this off hold now that the PR for error handling on manual requests is under review.

**Comment by neil-marcellini** [comment]
I'm starting on this now for the reason outlined [here](https://github.com/Expensify/App/issues/22126#issuecomment-1761608132) when closing the other issue.

**Comment by neil-marcellini** [comment]
On my plan for today

**Comment by neil-marcellini** [comment]
I'm demoting this to a weekly since I picked up a MEDIUM priority issue that will take precedence.

**Comment by neil-marcellini** [comment]
Never mind all that, I kept changing my mind lol

**Comment by neil-marcellini** [comment]
I'm setting a reminder to work on this tomorrow, or maybe today.

**Comment by neil-marcellini** [comment]
Ok I ran a little discussion [here](https://expensify.slack.com/archives/C05DWUDHVK7/p1699392640830819) in the slack room today and I think we just about nailed down the desired UX. However, we also decided to HOLD this on the [request money navigation refactor](https://github.com/Expensify/App/issues/26538), since so much is changing.

**Comment by neil-marcellini** [comment]
Still holding, but hopefully not next week!

**Comment by neil-marcellini** [comment]
I'm only working 50% and focusing on reviews and the P2P distance requests design, so I won't be able to make progress this week.

**Comment by neil-marcellini** [comment]
Also, please feel free to take this over if you like. I'll stay assigned since I plan to work on it eventually, but I won't be able to get to it for a while.

**Comment by neil-marcellini** [comment]
Up for grabs!

**Comment by dylanexpensify** [comment]
@stitesExpensify coming from [this message](https://expensify.slack.com/archives/C05DWUDHVK7/p1704227924440899?thread_ts=1704225983.057409&cid=C05DWUDHVK7), assigning a wave5 issue to ya! Feel free to switch it for a different one if needed though!

**Comment by stitesExpensify** [comment]
Sounds good! I'm all caught up from my OOO and will get started on this first thing next week

**Comment by stitesExpensify** [comment]
Read through the existing PR and asked a question of the contributors there. Will dive in further tomorrow!

**Comment by neil-marcellini** [comment]
I'm trying to finish off one other PR before I start on this.

**Comment by neil-marcellini** [comment]
Same story, but I'm going to set a reminder to get started on Wednesday.

**Comment by neil-marcellini** [comment]
I went back through the desired UX conversation and updated the description with the conclusions there. I'm going to need to do quite a bit of exploration on this one. I'm wondering if the current plan will work, or if this should be updated to use violations and create partial distance request transactions. It sounds more reasonable that way.

I'll explore this more soon and bring it up in wave9.

**Comment by neil-marcellini** [comment]
I'm checking on the priority of this issue in [Slack here](https://expensify.slack.com/archives/C03U7DCU4/p1707335658702659)

**Comment by neil-marcellini** [comment]
Based on that Slack conversation, we decided that this is [pretty low priority](https://expensify.slack.com/archives/C03U7DCU4/p1707336271298719?thread_ts=1707335658.702659&cid=C03U7DCU4) and we don't need anyone working on it right now. It has been moved to wave6 since wave5 has changed it's focus from working on the free plan, to deprecating it.

This issue is a small project that probably needs to go through the #whatsnext process, including a small design doc. You will have to decide if this should be implemented using an error handling approach, or using violations. Both are a decent amount of work. It's open for anyone to pick it up, and I’m happy to answer questions and give some guidance to whoever takes it on.

For now, once again, goodbye 👋
